src/auth/auth.service.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Injectable,
+  ConflictException,
+  InternalServerErrorException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import * as bcrypt from 'bcrypt';
+import { Model } from 'mongoose';
+import { JwtService } from '@nestjs/jwt';
+import { AuthRegisterInput } from './dto/auth-register.input';
+import { AuthLoginInput } from './dto/auth-login.input';
+import { User, UserDocument } from './models/user.model';
+import { JwtPayload } from './jwt-payload.interface';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectModel(User.name) private userModel: Model<UserDocument>,
+    private jwtService: JwtService,
+  ) {}
+
+  async register(input: AuthRegisterInput) {
+    const { email, password } = input;
+
+    const salt = await bcrypt.genSalt();
+    const hashedPassword = await bcrypt.hash(password, salt);
+
+    const user = new this.userModel({
+      email,
+      password: hashedPassword,
+    });
+
+    try {
```

Comment:
Tutaj bym zmienił logikę:
1. Szukam usera po mailu
2. Jeśli znajdę to rzucam błąd
3. Jeśli nie znajdę to dopiero wtedy dodaje

Czyli sprawdzam najpierw, a później dodaje do bazy zamiast dodawać i w try catch sprawdzać jaki błąd zwróciła baza.